### ISSUE TITLE
How do I alias my project root directory to tilde (~)?

### ISSUE THREAD

**Issue by douira** [jaredpalmer/backpack#166]
I'm using backpack to build my server for use together with programmatic use of a nuxt frontend.  In nuxt components and other files processed by nuxt, I can write
```
import thing from "~/util/thingFile"
```
and it resolves it to whatever path it needs to be to reach `util/thingFile`. How can I do the same thing in backpack? Can I extend the webpack config in `backpack.config.js` to achieve this?
I'm currently on the latest stable versions of nuxt and backpack.

**Comment by walleXD** [comment]
Just use the webpack config resolve entry

**Comment by douira** [comment]
@walleXD Could you elaborate with an example?

**Comment by daspete** [comment]
Simple backpack.config.js file

`const path = require('path')

const srcPath = path.resolve(process.cwd())

const aliases = {
    '~~seeds': `${ srcPath }/seeds`,
    '~~core': `${ srcPath }/core`,
    '~~stacks': `${ srcPath }/stacks`,
    '~~shared': `${ srcPath }/shared`
}

module.exports = {
    webpack: (config, options, webpack) => {
        config.resolve = {
            alias: aliases
        }

        return config
    }
}`

**Comment by douira** [comment]
Thanks, I'll test this later. It would be cool if there was an option to enable this automatically... Nuxt does this automatically for example. (see first comment)

**Comment by lednhatkhanh** [comment]
This doesn't work for me though

**Comment by daspete** [comment]
> This doesn't work for me though

What exactly does not work? What have your written in your config?

**Comment by douira** [comment]
This is my config now:
```
const path = require("path")

module.exports = {
  webpack: config => {
    //point to the server index
    config.entry.main = "./server/index.js"

    //setup aliases
    config.resolve = {
      alias: {
        "~~root": `${path.resolve(process.cwd())}`
      }
    }

    //return extended config
    return config
  }
}
```
When I write `import foo from '~/bar/foo.js'` it does not work and instead gives me a dependency not found error. What is the correct config or import to make this work?

**Comment by cassiolpaixao90** [comment]
Step 1: Install devdependecies 
 ```
    "babel-plugin-root-import": "^6.4.1",
    "eslint-import-resolver-babel-plugin-root-import": "^1.1.1",
    "eslint-plugin-import": "^2.18.2",
```

Step 2:  add in `.babelrc` in your project

```
"plugins": [
   [
      "babel-plugin-root-import",
      {
        "rootPathPrefix": "src",
        "rootPathSuffix": "src"
      }
    ]
]
```

Step 3:  create file `jsconfig.json` in root project

{
  "compilerOptions": {
    "baseUrl": "src",
    "paths": {
      "src/*": ["*"]
    }
  }
}

**Comment by douira** [comment]
@cassiolpaixao90 how do we use this if the root path is the root folder of the project? Do we use no `rootPathPrefix` or `rootPathPrefix: ""`?

**Comment by cassiolpaixao90** [comment]
> @cassiolpaixao90 Hey, by default when use backpack the folder `src` is `rootPathPrefix`

**Comment by cassiolpaixao90** [comment]
> @douira see this example ``https://github.com/cassiolpaixao90/example-backpack-alias``  =) ?

**Comment by douira** [comment]
Thats an interesting module. I read the npm page for `babel-plugin-root-import` and it does basically what I proposed in my [earlier comment](https://github.com/jaredpalmer/backpack/issues/166#issuecomment-569114179). Thanks for pointing out that there is a package that does this too. The question that I had asked you regarding the `rootPathPrefix` is answered by the documentation of the package in the [config section](https://www.npmjs.com/package/babel-plugin-root-import#config). The default config
```
"plugins": [
    [
      "babel-plugin-root-import",
      {
        "rootPathSuffix": "./",
        "rootPathPrefix": "~/"
      }
    ]
  ],
```
is what I was looking for. I'm going to see if I want to incorporate this into my projects as a plugin or the solution I wrote myself.
